src/components/CoverageSummary/CoverageSummary.tsx

```diff
@@ -0,0 +1,156 @@
+          <Typography sx={{ color: colors[1] }}>
+            Line Coverage: {summaryData[1]?.value}%
+          </Typography>
+          <Typography sx={{ color: colors[2] }}>
+            Method Coverage: {summaryData[2]?.value}%
+          </Typography>
+          <Typography sx={{ color: colors[3] }}>
+            Class Coverage: {summaryData[3]?.value}%
+          </Typography>
+        </Box>
+
+        <PieChart
+          series={[
+            {
+              data: summaryData.map((item, index) => ({
+                ...item,
+                color: colors[index],
+              })),
+              outerRadius: 50,
+              highlightScope: { faded: "global", highlighted: "item" },
+              faded: { innerRadius: 5, additionalRadius: -5 },
+            },
+          ]}
+          height={150}
+          width={150}
+          slotProps={{
+            legend: { hidden: true },
+          }}
+        />
+      </Box>
+
+      <Divider sx={{ bgcolor: "#ffffff", my: 2 }} />
+      <Typography variant="h6" sx={{ color: "#a892d4", mb: 1 }}>
+        Comparison with Base Branch ({baseBranchName})
+      </Typography>
+
+      <Box sx={{ display: "flex", gap: 4 }}>
+        <Box>
+          <Typography sx={{ color: colors[0] }}>
+            Coverage Change: {coverageChange > 0 ? "+" : ""}
+            {coverageChange}%{" "}
+            {coverageChange > 0 ? "(Improved)" : "(Decreased)"}
+          </Typography>
+          <Typography sx={{ color: colors[1] }}>
+            Line Coverage: {comparisonData[1]?.value}%
+          </Typography>
+
+          <Typography sx={{ color: "#64b5f6", mt: 2, mb: 1 }}>
+            Files Improved:
+          </Typography>
+          <List dense>
+            {improvedFiles.length > 0 ? (
+              improvedFiles.map((file, index) => (
+                <ListItem key={index} sx={{ py: 0 }}>
+                  <ListItemText
+                    primary={`• ${file}`}
+                    primaryTypographyProps={{ sx: { color: "white" } }}
+                  />
+                </ListItem>
+              ))
+            ) : (
+              <Typography sx={{ ml: 2 }}>• None</Typography>
+            )}
+          </List>
+
+          <Typography sx={{ color: "#64b5f6", mt: 1, mb: 1 }}>
+            Files Decreased Coverage:
+          </Typography>
+          <List dense>
+            {decreasedFiles.length > 0 ? (
+              decreasedFiles.map((file, index) => (
+                <ListItem key={index} sx={{ py: 0 }}>
+                  <ListItemText
+                    primary={`• ${file}`}
+                    primaryTypographyProps={{ sx: { color: "white" } }}
+                  />
+                </ListItem>
+              ))
+            ) : (
+              <Typography sx={{ ml: 2 }}>• None</Typography>
+            )}
+          </List>
+        </Box>
+
+        <PieChart
+          series={[
+            {
+              data: comparisonData.map((item, index) => ({
```

Comment:
This array mapping can be stored in a useMemo hook to avoid making the map each time component is re-render

Reply:
Done